Microsoft.Toolkit.Uwp.UI.Controls/ClassicMenu/ClassicMenu.Events.cs

```diff
@@ -0,0 +1,177 @@
+            {
+                if (args.VirtualKey == VirtualKey.Right)
+                {
+                    menuItem.ShowMenu();
+                    return true;
+                }
+
+                if (args.VirtualKey == VirtualKey.Up)
+                {
+                    var currentMenuItemIndex = menu.Items.IndexOf(menuItem);
+                    if (currentMenuItemIndex > 0)
+                    {
+                        FocusManager.TryMoveFocus(FocusNavigationDirection.Up);
+                        return true;
+                    }
+                }
+                else if (args.VirtualKey == VirtualKey.Down)
+                {
+                    var currentMenuItemIndex = menu.Items.IndexOf(menuItem);
+                    if (currentMenuItemIndex < menu.Items.Count - 1)
+                    {
+                        FocusManager.TryMoveFocus(FocusNavigationDirection.Down);
+                        return true;
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        private static string MapInputToGestureKey(KeyEventArgs args)
+        {
+            var isCtrlDown = Window.Current.CoreWindow.GetKeyState(VirtualKey.Control).HasFlag(CoreVirtualKeyStates.Down);
+            var isShiftDown = Window.Current.CoreWindow.GetKeyState(VirtualKey.Shift).HasFlag(CoreVirtualKeyStates.Down);
+
+            if (!isCtrlDown && !isShiftDown)
+            {
+                return null;
+            }
+
+            StringBuilder gestureKeyBuilder = new StringBuilder();
+
+            if (isCtrlDown)
+            {
+                gestureKeyBuilder.Append(CtrlValue);
+                gestureKeyBuilder.Append("+");
+            }
+
+            if (isShiftDown)
+            {
+                gestureKeyBuilder.Append(ShiftValue);
+                gestureKeyBuilder.Append("+");
+            }
+
+            if (args.VirtualKey == VirtualKey.None)
+            {
+                gestureKeyBuilder.Remove(gestureKeyBuilder.Length - 1, 1);
+            }
+            else
+            {
+                gestureKeyBuilder.Append(args.VirtualKey);
+            }
+
+            return gestureKeyBuilder.ToString();
+        }
+
+        private void ClassicMenu_Loaded(object sender, RoutedEventArgs e)
+        {
+            var wrapPanel = ItemsPanelRoot as WrapPanel.WrapPanel;
+            if (wrapPanel != null)
+            {
+                wrapPanel.Orientation = Orientation;
+            }
+
+            Dispatcher.AcceleratorKeyActivated -= Dispatcher_AcceleratorKeyActivated;
+            Window.Current.CoreWindow.KeyDown -= CoreWindow_KeyDown;
+            Dispatcher.AcceleratorKeyActivated += Dispatcher_AcceleratorKeyActivated;
+            Window.Current.CoreWindow.KeyDown += CoreWindow_KeyDown;
+        }
+
+        private void ClassicMenu_Unloaded(object sender, RoutedEventArgs e)
+        {
+            Dispatcher.AcceleratorKeyActivated -= Dispatcher_AcceleratorKeyActivated;
```

Comment:
I guess it should be done automatically ? Once the page is unloaded all events, properties should destroyed !!
Regarding the general events like Dispatcher.AcceleratorKeyActivated it make sense to unsubscribe at unloaded but does unsubscribe from unload event here helps ? :)